front-end/src/pages/EditEventDetails.jsx

```diff
@@ -110,260 +65,70 @@ export default function EditEventDetails() {
     }
   };
 
-  // update event details in BE using utility function
-  const updateEvent = async () => {
-    // adding check to see if virtualEventLink is blank
-
-    let responseStatus = await updateEventDetails(
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    const responseStatus = await updateEventDetails(
       eventID,
       title,
       eventStart,
       eventEnd,
       timeZone,
       eventType,
-      eventVenue || "", // Ensure non-null
-      eventVenueAddress || "", // Ensure non-null
-      description || "", // Ensure non-null
+      eventVenue,
+      eventVenueAddress,
+      description,
       category,
-      eventPhoto, // Already adjusted to send as base64 string
-      virtualEventLink, 
-      location,
-      eventCoordinates,
+      eventPhoto,
+      virtualEventLink,
+      location, 
+      eventCoordinates
     );
     if (responseStatus) {
-      navigate("/profile");
+      navigate('/profile');
     }
   };
```

Comment:
This would be great for a post request but for edit requests if anything is empty it will overwrite data already there with empty strings. It looks like you're pulling data before doing this so it should be fine it's just something to look out for.

Reply:
I was thinking that would be alright, like if the user deletes out a photo or something and decides they don't want one kinda thing or etc. But we can totally switch it back!